### PR TITLE
add st-link_cli flashing

### PR DESCRIPTION
**Pull request by hsgw**
Added st-link_cli SWD flashing.

But I don't add st-link_cli to install script, because it needs registlation to download.

### CODE REVIEW

**Comment by jackhumbert** [comment]
Nice - would it make sense to have it formatted like `ST_LINK_CLI` and `st-link-cli` to match the other notations (eg `dfu-util`) we have?

**Comment by hsgw** [comment]
Hi, I adjusted notations.

File: tmk_core/chibios.mk [hunk 1]
```diff
@@ -198,10 +198,13 @@ ifneq ("$(SERIAL)","")
 	DFU_ARGS += -S $(SERIAL)
 endif
 
+ST_LINK_ARGS ?=
+
 # List any extra directories to look for libraries here.
 EXTRALIBDIRS = $(RULESPATH)/ld
 
 DFU_UTIL ?= dfu-util
+ST_LINK_CLI ?= st-link_cli
```
Comment (jackhumbert) [inline tmk_core/chibios.mk]: Could you adjust `st-link_cli` to be `st-link-cli` here as well?

Comment (hsgw) [reply tmk_core/chibios.mk]: This line is setting command or path.
We can't modify.

Comment (drashna) [reply tmk_core/chibios.mk]: Well, rather, it looks like it's a file name, basically, like dfu-util.

In fact, downloaded this on my system (windows), and it's explicitly named this, actually: `ST-LINK_CLI.exe`. 

So this should be fine.

Comment (jackhumbert) [reply tmk_core/chibios.mk]: Ah, ok. Thanks!

**Comment by jackhumbert** [comment]
Cool! Thanks :)

**Pull request merged**